io/include/traccc/io/read_measurements.hpp

```diff
@@ -28,12 +28,14 @@ namespace traccc::io {
 /// @param[out] measurements The measurement collection to fill
 /// @param[in]  event     The event ID to read in the measurements for
 /// @param[in]  directory The directory holding the measurement data files
-/// @param[in]  dd        The detector description to point the measurements at
+/// @param[in]  use_acts_geom_source  Use acts geometry source
+/// @param[in]  detector  detray detector
 /// @param[in]  format    The format of the measurement data files (to read)
 ///
 void read_measurements(measurement_collection_types::host& measurements,
                        std::size_t event, std::string_view directory,
-                       const silicon_detector_description::host* dd = nullptr,
+                       bool use_acts_geom_source = true,
+                       const traccc::default_detector::host* detector = nullptr,
```

Comment:
🤔 I think the move here is to replace `silicon_detector_description::host` with `default_detector::host`, and that's it. The additional boolean is completely unnecessary as far as I can see.
  - When it's set to `true`, there must be a valid `detector` pointer;
  - When it's set to `false`, we don't care about the `detector` pointer.

So we should just use the same logic that's there at the moment.
  - When `detector` is specified, it's used to re-map the geometry IDs;
  - When it's left as `nullptr`, no re-mapping is done.

Reply:
> One can still argue that we can try to check whether there is an ACTS geom source or not in the detray file

Actually this is not the case because  we should be able to run the detray simulation on arbitrary geometry file. Yeah.. so I think `use_acts_geom_source` is necessary in any case

Reply:
Same here